src/action.c

```diff
@@ -158,6 +169,30 @@ action_arg_add_int(struct action *action, const char *key, int value)
 	wl_list_append(&action->args, &arg->base.link);
 }
 
+void
+action_arg_add_qlist(struct action *action, const char *key)
+{
+	assert(action);
+	assert(key);
+	struct action_arg_list *arg = znew(*arg);
+	arg->base.type = LAB_ACTION_ARG_QUERY_LIST;
+	arg->base.key = xstrdup(key);
+	wl_list_init(&arg->value);
+	wl_list_append(&action->args, &arg->base.link);
+}
+
+void
+action_arg_add_alist(struct action *action, const char *key)
+{
+	assert(action);
+	assert(key);
+	struct action_arg_list *arg = znew(*arg);
+	arg->base.type = LAB_ACTION_ARG_ACTION_LIST;
+	arg->base.key = xstrdup(key);
+	wl_list_init(&arg->value);
+	wl_list_append(&action->args, &arg->base.link);
+}
```

Comment:
Suggest combining these two into 3 functions:
Internal `static void action_arg_add_list(struct action *action, const char *key, enum action_arg_type)` doing most of the work and two wrappers for the new types.

Reply:
Agree, done.